libs/ledger/tests/unit/storage/transaction_generator.hpp

```diff
@@ -0,0 +1,96 @@
+#pragma once
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "chain/address.hpp"
+#include "chain/transaction.hpp"
+#include "chain/transaction_builder.hpp"
+#include "core/bitvector.hpp"
+#include "core/byte_array/byte_array.hpp"
+#include "core/byte_array/const_byte_array.hpp"
+#include "core/random/lcg.hpp"
+#include "crypto/ecdsa.hpp"
+
+class TransactionGenerator
+{
+public:
+  using Transaction        = fetch::chain::Transaction;
+  using TransactionBuilder = fetch::chain::TransactionBuilder;
+  using TransactionPtr     = TransactionBuilder::TransactionPtr;
+  using Txs                = std::vector<TransactionGenerator::TransactionPtr>;
+
+  TransactionPtr operator()()
+  {
+    return TransactionBuilder{}
+        .From(address_)
+        .ValidUntil(1000)
+        .TargetChainCode("foo.bar.baz", BitVector{})
+        .Action("test")
+        .Data(GenerateRandomData())
+        .Signer(public_key_)
+        .Seal()
+        .Sign(private_key_)
+        .Build();
+  }
+
+  Txs GenerateRandomTxs(std::size_t count)
```

Comment:
I recall there are other fns floating around that generate txs - can you delete them?

Reply:
Not really - I did look into them but they seemed to be much more complicated than the above which is why I opted to have this simple subset for these tests